src/integrant/repl.clj

```diff
@@ -29,59 +37,110 @@
           (throw (wrap-ex ex halt-ex))))
       (throw ex))))
 
-(defn- init-system [config]
-  (build-system
-   #(ig/init config)
-   #(ex-info "Config failed to init; also failed to halt failed system"
-             {:init-exception %1}
-             %2)))
-
-(defn- resume-system [config system]
-  (build-system
-   #(ig/resume config system)
-   #(ex-info "Config failed to resume; also failed to halt failed system"
-             {:resume-exception %1}
-             %2)))
-
-(defn init []
-  (alter-var-root #'state/system (fn [sys]
-                                   (halt-system sys)
-                                   (init-system state/config)))
-  :initiated)
-
-(defn go []
-  (prep)
-  (init))
-
-(defn clear []
-  (alter-var-root #'state/system (fn [sys] (halt-system sys) nil))
-  (alter-var-root #'state/config (constantly nil))
-  :cleared)
-
-(defn halt []
-  (halt-system state/system)
-  (alter-var-root #'state/system (constantly nil))
-  :halted)
-
-(defn suspend []
-  (when state/system (ig/suspend! state/system))
-  :suspended)
-
-(defn resume []
-  (if-let [prep state/preparer]
-    (let [cfg (prep)]
-      (alter-var-root #'state/config (constantly cfg))
-      (alter-var-root #'state/system (fn [sys]
-                                       (if sys
-                                         (resume-system cfg sys)
-                                         (init-system cfg))))
-      :resumed)
-    (throw (prep-error))))
-
-(defn reset []
-  (suspend)
-  (repl/refresh :after 'integrant.repl/resume))
-
-(defn reset-all []
-  (suspend)
-  (repl/refresh-all :after 'integrant.repl/resume))
+(defn- init-system
+  ([config]
+   (init-system config (keys config)))
+
+  ([config keys]
+   (build-system
+    #(ig/init config keys)
+    #(ex-info "Config failed to init; also failed to halt failed system"
+              {:init-exception %1}
+              %2))))
+
+(defn- resume-system
+  ([config system]
+   (resume-system config system (keys config)))
+
+  ([config system keys]
+   (build-system
+    #(ig/resume config system keys)
+    #(ex-info "Config failed to resume; also failed to halt failed system"
+              {:resume-exception %1}
+              %2))))
+
+(defn init
```

Comment:
Something more like:

```clojure
(defn init
  ([] (init nil))
  ([keys]
   (alter-var-root #'state/system (fn [sys]
                                    (halt-system sys keys)
                                    (init-system state/config keys)))
```

Then `halt-system` and `init-system` can check for the presence of `nil`.